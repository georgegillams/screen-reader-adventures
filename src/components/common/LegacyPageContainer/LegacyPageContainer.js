import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { cssModules } from 'gg-components/helpers/cssModules';

import STYLES from './page-container.scss';
import { DEBUG_SHOW_PAGE_CONTAINER_KEY } from 'helpers/storageConstants';

export const LAYOUT_STYLES = {
  default: 'default',
  defaultCenter: 'defaultCenter',
  prose: 'prose',
  proseCenter: 'proseCenter',
  fullWidth: 'fullWidth',
  fullWidthCenter: 'fullWidthCenter',
};

const getClassName = cssModules(STYLES);

const LegacyPageContainer = props => {
  const { className, layout, bottomPadding, ...rest } = props;
  const [showDebug, setShowDebug] = useState(false);

  useEffect(() => {
    setShowDebug(window.localStorage.getItem(DEBUG_SHOW_PAGE_CONTAINER_KEY) === 'true');
  }, []);

  const outerClassNames = [getClassName('page-container__container')];
  if (bottomPadding) {
    outerClassNames.push([getClassName('page-container__container--bottom-padding')]);
  }
  if (showDebug) {
    outerClassNames.push([getClassName('page-container__container--debug')]);
  }
  if (layout === LAYOUT_STYLES.default || layout === LAYOUT_STYLES.defaultCenter) {
    outerClassNames.push([getClassName('page-container__container--regular')]);
  }
  if (layout === LAYOUT_STYLES.prose || layout === LAYOUT_STYLES.proseCenter) {
    outerClassNames.push([getClassName('page-container__container--prose')]);
  }
  if (layout === LAYOUT_STYLES.fullWidth || layout === LAYOUT_STYLES.fullWidthCenter) {
    outerClassNames.push([getClassName('page-container__container--full-width')]);
  }
  if (
    layout === LAYOUT_STYLES.defaultCenter ||
    layout === LAYOUT_STYLES.proseCenter ||
    layout === LAYOUT_STYLES.fullWidthCenter
  ) {
    outerClassNames.push([getClassName('page-container__container--center')]);
  }

  if (className) {
    outerClassNames.push(className);
  }

  return <main id="main" className={outerClassNames.join(' ')} {...rest} />;
};

LegacyPageContainer.propTypes = {
  layout: PropTypes.oneOf(Object.keys(LAYOUT_STYLES)),
  className: PropTypes.string,
  bottomPadding: PropTypes.bool,
};

LegacyPageContainer.defaultProps = {
  layout: LAYOUT_STYLES.default,
  className: null,
  bottomPadding: true,
};

export default LegacyPageContainer;
