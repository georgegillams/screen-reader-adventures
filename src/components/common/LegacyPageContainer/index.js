export { default, LAYOUT_STYLES } from './LegacyPageContainer';
