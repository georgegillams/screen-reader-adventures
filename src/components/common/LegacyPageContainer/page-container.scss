@import '~utils/tokens';

.page-container {
  &__container {
    display: flow-root;
    width: 100%;
    margin-right: auto;
    margin-left: auto;
    // overflow is needed to ensure elements animated
    // from outside the page don't affect page width
    overflow: hidden;

    &--bottom-padding {
      padding-bottom: $spacing-lg;
    }

    &--debug {
      background-color: orange;

      @media (prefers-color-scheme: dark) {
        background-color: darkorange;
      }
    }

    &--regular {
      max-width: $page-max-width;
      padding-right: $spacing-lg;
      padding-left: $spacing-lg;

      @include bpk-breakpoint-tablet {
        padding-right: $spacing-base;
        padding-left: $spacing-base;
      }
    }

    &--prose {
      max-width: $page-max-width-prose;
      padding-right: $spacing-lg;
      padding-left: $spacing-lg;

      @include bpk-breakpoint-tablet {
        padding-right: $spacing-base;
        padding-left: $spacing-base;
      }
    }

    &--center {
      display: flex;
      width: 100%;
      flex-direction: column;
      align-items: center;
      text-align: center;
    }
  }
}
