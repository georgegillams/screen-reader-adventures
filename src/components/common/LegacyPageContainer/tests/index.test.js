import React from 'react';

import { render } from '@testing-library/react';

import LegacyPageContainer, { LAYOUT_STYLES } from '..';

describe('<LegacyPageContainer />', () => {
  const children = <div>some text</div>;

  it('Should render correctly', () => {
    const { container } = render(<LegacyPageContainer>{children}</LegacyPageContainer>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with custom className', () => {
    const { container } = render(
      <LegacyPageContainer className={'test-custom-classname'}>{children}</LegacyPageContainer>
    );

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with prose', () => {
    const { container } = render(<LegacyPageContainer layout={LAYOUT_STYLES.prose}>{children}</LegacyPageContainer>);

    expect(container).toMatchSnapshot();
  });

  it('Should render correctly with defaultCenter', () => {
    const { container } = render(
      <LegacyPageContainer layout={LAYOUT_STYLES.defaultCenter}>{children}</LegacyPageContainer>
    );

    expect(container).toMatchSnapshot();
  });
});
