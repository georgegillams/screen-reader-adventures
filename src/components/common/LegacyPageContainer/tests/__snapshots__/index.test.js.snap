// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LegacyPageContainer /> Should render correctly 1`] = `
<div>
  <main
    class="page-container__container page-container__container--bottom-padding page-container__container--regular"
    id="main"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;

exports[`<LegacyPageContainer /> Should render correctly with custom className 1`] = `
<div>
  <main
    class="page-container__container page-container__container--bottom-padding page-container__container--regular test-custom-classname"
    id="main"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;

exports[`<LegacyPageContainer /> Should render correctly with defaultCenter 1`] = `
<div>
  <main
    class="page-container__container page-container__container--bottom-padding page-container__container--regular page-container__container--center"
    id="main"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;

exports[`<LegacyPageContainer /> Should render correctly with prose 1`] = `
<div>
  <main
    class="page-container__container page-container__container--bottom-padding page-container__container--prose"
    id="main"
  >
    <div>
      some text
    </div>
  </main>
</div>
`;
